openthread_border_router/Dockerfile

```
@@ -88,7 +89,7 @@ RUN \
         -DOT_COAP=OFF \
         -DOT_COAPS=OFF \
         -DOT_DNS_CLIENT_OVER_TCP=OFF \
-        -DOT_THREAD_VERSION=1.3 \
+        -DOT_THREAD_VERSION=1.4 \
```

From what I can tell, this mainly enables the ePSKc which is an ephemeral variant of the PSKc, which is used for the Thread specific commissioning. This will be used for credentials sharing, but we don't implement that yet.

Other features behind the flag are Thread domain name, which from what I understand also only becomes relevant if there are other Thread 1.4 BRs.

Setting this to 1.4 will advertise us as an Thread 1.4 BR. I don't think is problematic currently, but maybe it becomes once other systems rely on e.g. credential sharing to work. For now I prefer to stay with 1.3.

```suggestion
        -DOT_THREAD_VERSION=1.3 \
```